great_expectations/data_context/store/ge_cloud_store_backend.py

```diff
@@ -426,11 +484,19 @@ def remove_key(self, key):
             response = requests.delete(
                 url, json=data, headers=self.headers, timeout=self.TIMEOUT
             )
-            response_status_code = response.status_code
-
-            if response_status_code < 300:
-                return True
+            response.raise_for_status()
+            return True
+        except requests.HTTPError as http_exc:
+            # TODO: GG 20220819 should we raise an error here instead of returning False
+            logger.warning(
+                f"Unable to delete object in GE Cloud Store Backend: {_get_user_friendly_error_message(http_exc)}"
+            )
             return False
+        except requests.Timeout as timeout_exc:
+            logger.exception(timeout_exc)
+            raise StoreBackendError(
+                "Unable to delete object in GE Cloud Store Backend: This is likely a transient error. Please try again."
+            )
         except Exception as e:
             logger.debug(str(e))
```

Comment:
Just a thought and non blocking - looks like some repeated code, any way to make this more dry? Maybe we can pull this error handling out into a handler that takes a callable and the error messages? Callable for the logic in the `try:`?

Reply:
Yes, I have a TODO in here that I'd like to extract all the HTTP stuff into a dedicated client/service.
I plan that as a followup.